pkg/envoy/ingress_translator.go

```diff
@@ -311,6 +321,13 @@ func translateIngresses(ingresses []*k8s.Ingress, syncSecrets bool, secrets []*v
 					}
 				}
 
+				if i.Annotations["yggdrasil.uswitch.com/weight"] != "" {
+					weight64, err := strconv.ParseUint(i.Annotations["yggdrasil.uswitch.com/weight"], 10, 32)
+					if err == nil {
+						envoyIngress.setWeight(uint32(weight64))
+					}
+				}
```

Comment:
This sets the same weight to all the cluster's upstreams.  
What you want is to set it independently for each upstream (each k8s ingress resource) otherwise it has no effect at all.